Added support for replacing style with a RTL version in the add_edito…

pbearne opened the pull request:
Added support for replacing a style sheet with a RTL version in the add_editor_style function

Added rtl editor styles sheets for twentytwentytOne theme
Added tests for add_editor_style

Trac ticket: https://core.trac.wordpress.org/ticket/52294

pbearne committed:
Added support for replacing style with a RTL version in the add_editor_style function
Added rtl editor styles sheets for twentytwentytOne theme
Added tests for add_editor_style

JustinyAhin commented:
Hello @pbearne, could you add some tests instructions to the PR please?

pbearne commented:
testing:
install a right to left lang
open the editor and note the lay does match the front end

Apply patch and the layout should match as we are not now loading the ltr CSS just the rtl CSS

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,123 @@
+class Test_Add_Editor_Style extends WP_UnitTestCase {
+	/**
+	 * clear out the editor styles.
+	 *
+	 * @return void
```
These lines aren't necessary and can be removed for all methods in the PR.

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,123 @@
+
+class Test_Add_Editor_Style extends WP_UnitTestCase {
+	/**
+	 * clear out the editor styles.
```
```suggestion
	 * Reset the editor styles to an empty array.
```

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	/**
+	 * run the default settings.
```
```suggestion
	 * Tests the default editor styles.
```

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,123 @@
+	 * @return void
+	 */
+	public function set_up() {
+		parent::set_up(); // TODO: Change the autogenerated stub
```
Is this `TODO:` still necessary?

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	/**
+	 * run default setting in rtl mode.
```
```suggestion
	 * Tests the default editor styles in RTL mode.
```

costdev replied:
This was marked resolved with no change and no reply - could you let me know your thoughts?

pbearne committed:
Merge branch 'WordPress:trunk' into 52294_add_option_to_replace_editer_css_for_rtl

pbearne committed:
changes from review

audrasjb commented on src/wp-includes/theme.php:
```diff
@@ -2079,24 +2079,41 @@ function wp_update_custom_css_post( $css, $args = array() ) {
+ * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root.
+ *                                 Defaults to 'editor-style.css'
+ *
+ * @param string $rtl_mode Optional. Whether if the site in RTL mode. Should the RTL style replace the default sheet or be added as well'.
```
DocBlock issue: This docblock line should be consistently indented with the other param.

audrasjb commented on src/wp-includes/theme.php:
```diff
@@ -2079,24 +2079,41 @@ function wp_update_custom_css_post( $css, $args = array() ) {
+ *
+ * @param string $rtl_mode Optional. Whether if the site in RTL mode. Should the RTL style replace the default sheet or be added as well'.
+ *                         Set to 'replace' to replace the sheet.
+ *                         Defaults to 'add'. @sine 5.9.0
```
"sine" should be replaced with "since". Also, the `since` mention should be moved right after `@since 3.0.0`.

pbearne replied:
doc block updated

audrasjb reviewed (changes requested):
Thanks for the PR! 
I added two small change requests :)

pbearne committed:
Merge branch 'WordPress:trunk' into 52294_add_option_to_replace_editer_css_for_rtl

pbearne committed:
doc block update

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+ * @covers ::add_editor_style
+ */
+
+class Test_Add_Editor_Style extends WP_UnitTestCase {
```
```suggestion
class Tests_Theme_addEditorStyle extends WP_UnitTestCase {
```

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+
+	/**
+	 * Tests the default editor styles in RTL mode.
+	 *
```
Should have `@ticket` annotation.

🔢 Applies to all test methods.

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @group themes
+ * @ticket 52294
```
`@ticket` annotations aren't needed at the class level.

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+		unset( $_wp_theme_features['editor-style'] );
+		add_editor_style();
+
+		$this->AssertContains( 'editor-style', array_keys( $_wp_theme_features ) );
```
```suggestion
		$this->assertContains( 'editor-style', array_keys( $_wp_theme_features ) );
```

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+	 *
+	 * @return void
+	 */
+	public function test_add_editor_style_sets_theme_support() {
+		global $_wp_theme_features;
+		unset( $_wp_theme_features['editor-style'] );
```
Is `$_wp_theme_features` reset to its previous value after each test? If not, then this should be reset to its previous value in the test method, before the assertion.

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+	/**
+	 * Tests the default editor styles in RTL mode.
+	 *
+	 * @return void
```
This return annotation isn't necessary.

🔢 Applies to other instances in the PR.

costdev commented on tests/phpunit/tests/theme/addEditorStyle.php:
```diff
@@ -0,0 +1,121 @@
+	}
+
+	/**
+	 * run default setting in rtl mode.
```
Docblock summaries should begin with a capital letter and use third-person singular verbs (i.e. "Runs", not "run"). Tests, in particular, should begin with "Tests that X does Y when Z", or similar.

🔢 Applies to other instances in the PR.

costdev commented on src/wp-includes/theme.php:
```diff
@@ -2096,24 +2096,38 @@ function wp_update_custom_css_post( $css, $args = array() ) {
  * Since version 3.4 the TinyMCE body has .rtl CSS class.
  * It is a better option to use that class and add any RTL styles to the main stylesheet.
  *
+ * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root. Defaults to 'editor-style.css'.
+ *
+ * @param string $rtl_mode Optional. Whether if the site in RTL mode. Should the RTL style replace the default sheet or be added as well. Set to 'replace' to replace the sheet.
```
These should not have been moved. They should be at the bottom of the docblock, with no empty line between them.

pbearne replied:
fixed

costdev commented on src/wp-includes/theme.php:
```diff
@@ -2096,24 +2096,38 @@ function wp_update_custom_css_post( $css, $args = array() ) {
+ * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root. Defaults to 'editor-style.css'.
+ *
+ * @param string $rtl_mode Optional. Whether if the site in RTL mode. Should the RTL style replace the default sheet or be added as well. Set to 'replace' to replace the sheet.
+ *
+ * @return void
+ *
```
This annotation is not necessary.

pbearne replied:
removed

costdev commented on src/wp-includes/theme.php:
```diff
@@ -2096,24 +2096,38 @@ function wp_update_custom_css_post( $css, $args = array() ) {
+ * @return void
+ *
  * @since 3.0.0
+ * @since 5.9.0 Defaults to 'add'.
```
```suggestion
 * @since 6.1.0 Added the `$rtl_mode` parameter. Defaults to 'add'.
```

costdev commented on src/wp-includes/theme.php:
```diff
@@ -2096,24 +2096,38 @@ function wp_update_custom_css_post( $css, $args = array() ) {
 		$rtl_stylesheet = str_replace( '.css', '-rtl.css', $stylesheet[0] );
-		$stylesheet[]   = $rtl_stylesheet;
+
+		if ( 'replace' === strtolower( trim( $rtl_mode ) ) ) {
```
`$rtl_mode` is not type-safe. Need to make sure `trim()` and `strtolower()` are being used on a `string`.

pbearne replied:
fix in my version

costdev reviewed (changes requested):
Thanks for working on this Paul! I've left some thoughts below.

pbearne committed:
Update tests/phpunit/tests/theme/addEditorStyle.php

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

pbearne committed:
Update tests/phpunit/tests/theme/addEditorStyle.php

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

pbearne committed:
Update src/wp-includes/theme.php

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

pbearne committed:
fixed whitespace

costdev commented on src/wp-includes/theme.php:
```diff
@@ -2097,23 +2097,33 @@ function wp_update_custom_css_post( $css, $args = array() ) {
  *
- * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root.
- *                                 Defaults to 'editor-style.css'
+ * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root. Defaults to 'editor-style.css'.
+ * @param string $rtl_mode Optional. Whether if the site in RTL mode. Should the RTL style replace the default sheet or be added as well. Set to 'replace' to replace the sheet.
```
```suggestion
 * @param array|string $stylesheet Optional. Stylesheet name or array thereof, relative to theme root.
 *                                 Defaults to 'editor-style.css'.
 * @param string       $rtl_mode   Optional. When the site is in RTL mode, whether the RTL stylesheet should replace the default stylesheet, or be added as well.
 *                                 Accepts 'replace' or 'add'. Defaults to 'add'.
```
- Aligned the parameter names and descriptions.
- Updated the description of `$rtl_mode` for clarity and to show the default value.
- Restored `Defaults to 'editor-style.css'.` to its original position on a new line.

pbearne committed:
Update src/wp-includes/theme.php

Co-authored-by: Colin Stewart <79332690+costdev@users.noreply.github.com>

pbearne committed:
Merge branch 'trunk' into 52294_add_option_to_replace_editer_css_for_rtl

pbearne committed:
Improve RTL mode validation in theme.php

In the 'theme.php' script, added a conditional check to ensure the 'rtl_mode' is a string before performing a comparison. The code for the PHP unit tests has also been tidied up and enhanced the documentation around each test function to explicitly mention the ticket number related to the test.

pbearne committed:
Make test function in addEditorStyle.php public

Modified the test function 'test_add_editor_style_custom_path' in 'addEditorStyle.php' to make it public, enabling external access. This also improves overall code tidiness and readability.

pbearne committed:
Update addEditorStyle.php
